### ISSUE TITLE
Version lock bun to lower than 1.0.4 for now

### ISSUE THREAD

**Issue by frankcalise** [infinitered/ignite#2522]
## Problem

`bun test` doesn't pass on version 1.0.5 after a project is generated, however it does on 1.0.3. This was detected on ci in #2513 

## Enhancement 

Update the cli to version check for bun if it's used as the package manager to help manage the user's expectations until they fix the issue in bun.

It would be great to add bun version to the `doctor` command as well

## Related Issues
https://github.com/oven-sh/bun/issues/4259

**Comment by markrickert** [comment]
The workaround listed in this comment works if you are running `bun` > 1.0.4 in an ignited app: https://github.com/oven-sh/bun/issues/4259#issuecomment-1726784398

Or to downgrade your local `bun` installation, you have to [uninstall](https://bun.sh/docs/installation#uninstall) and then reinstall with this command: `curl -fsSL https://bun.sh/install | bash -s "bun-v1.0.4"`

**Comment by frankcalise** [comment]
Might not be needed after 1.0.6 lands: https://github.com/oven-sh/bun/issues/6421

But we should still add the version to doctor so we can catch if people have a lower version than latest

**Comment by jamonholmgren** [comment]
Relevant: https://x.com/jarredsumner/status/1712287791995060642?s=46&t=WOd3LsbkVkwKLjH5qFAnLg

**Comment by Jarred-Sumner** [comment]
Let me know if you are still running into issues

**Comment by frankcalise** [comment]
@Jarred-Sumner hey still seeing issues with 1.0.6 installing via bun now as well. If I drop back down to 1.0.4, it works out.

`npx ignite-cli@next new PizzaAppNextBun --packager=bun --verbose --overwrite --yes` results in:

```
    ⠸ Installing bun dependencies (wow these are heavy)
/Users/fcalise/.npm/_npx/2351a7c35afcf833/node_modules/gluegun/build/index.js:15
    throw up;
    ^
bun install v1.0.6 (969da088)
 Resolving dependencies
 Resolved, downloaded and extracted [1158]
 Saved lockfile
/bin/bash: /react-dompiz: No such file or directory

(Use `node --trace-uncaught ...` to show where the exception was thrown)

Node.js v18.11.0
```

**Comment by Jarred-Sumner** [comment]
We’ve identified the cause and will have a fix shortly

**Comment by Jarred-Sumner** [comment]
@frankcalise can you give it another try on v1.0.7? This should be fixed along with more changes related to peer dependencies.

**Comment by frankcalise** [comment]
@Jarred-Sumner yup, bun install works again now! thanks.

`bun test` will not complete due to not having all the jest matcher coverage, but for now we'll roll with `bun run test` and it'll work out

**Comment by frankcalise** [comment]
@Jarred-Sumner as far as running our RN app, I have to add a `bunfig.toml` and disable peer installs for things to run correctly

**Comment by Jarred-Sumner** [comment]
> @Jarred-Sumner as far as running our RN app, I have to add a `bunfig.toml` and disable peer installs for things to run correctly

This is not great. We are working on porting more tests from other package managers to bun install and hopefully that will more thoroughly catch this kind of bug. 

For now, one thing you can try is removing your lockfile. `rm -rf bun.lockb node_modules ~/.bun/install && bun install`

**Comment by frankcalise** [comment]
@Jarred-Sumner So that worked, but the process took about 900 _seconds_, but `yarn ios` boots up without errors now and no `bunfig.toml` present. 

<img width="444" alt="image" src="https://github.com/infinitered/ignite/assets/374022/abc0d1c1-86af-4553-8c3e-156422f17be6">

Afterwards, I deleted `node_modules` again and just ran `bun install` and it completed in 1100 ms

**Comment by jamonholmgren** [comment]
Probably related: I had this error:

```
 ERROR  Invariant Violation: View config getter callback for component `RNCSafeAreaProvider` must be a function (received `undefined`).
 ```

<img width="866" alt="CleanShot 2023-10-30 at 18 33 44@2x" src="https://github.com/infinitered/ignite/assets/1479215/46482d92-cff9-4fe6-9c55-47d9c83bd8f3">

 
 When I ran the above:
 
 ```
 rm -rf bun.lockb node_modules ~/.bun/install && bun install
 ```
 
 it took about 43 seconds, and then it worked fine after that.
 
 
<img width="862" alt="CleanShot 2023-10-30 at 18 37 40@2x" src="https://github.com/infinitered/ignite/assets/1479215/f2f5e2a3-6bef-4bb0-b4ea-005c869dff9a">

**Comment by janwiebe-jump** [comment]
@jamonholmgren Could this be related to the peer-depencencies bug that has been fixed in [bun 1.0.8]?(https://bun.sh/blog/bun-v1.0.8#bun-install-peer-dependency-duplicate-version-bugfix)

**Comment by frankcalise** [comment]
Closing this as bun v1.0.10 no longer has this issue (maybe previous versions but I was coming off 1.0.7)

Will remove the bunfig from v9 which disabled peer install in a PR
